qt/widgets/common/src/FitScriptGeneratorModel.cpp

```diff
@@ -99,10 +100,10 @@ FitScriptGeneratorModel::~FitScriptGeneratorModel() {
 
 void FitScriptGeneratorModel::subscribePresenter(IFitScriptGeneratorPresenter *presenter) { m_presenter = presenter; }
 
-void FitScriptGeneratorModel::removeWorkspaceDomain(std::string const &workspaceName, WorkspaceIndex workspaceIndex) {
-  auto const removeIter = findWorkspaceDomain(workspaceName, workspaceIndex);
-  if (removeIter != m_fitDomains.cend()) {
-    m_fitDomains.erase(removeIter);
+void FitScriptGeneratorModel::removeDomain(FitDomainIndex domainIndex) {
+  auto const removeIter = m_fitDomains.begin() + domainIndex.value;
+  if (removeIter < m_fitDomains.cend()) {
```

Comment:
This doesn't look correct to me, we should check that `m_fitDomains.size() > domainIndex.value` before doing the add, as going past cend() can be [undefined behaviour](https://stackoverflow.com/a/39820063) and can be a trap for future developers

``` c++
if domainIndex.value > m_fitDomains.size()
    return;

auto const removeIter = m_fitDomains.begin() + domainIndex.value;
if (removeIter != m_fitDomains.cend())
```